### ISSUE TITLE
Shadowfiend, priest pet is bugged.

### ISSUE THREAD

**Issue by mirailol** [SunwellTracker/issues#3371]
Decription: i have noticed a bug with priests pet, shadowfiend. 

How it works: when i use my pet i noticed that it can be kited by any class that can slow a target, and that the pet doesn't move fast enough. it can be distanced by any player that has nitro boots or speed potions. and priests go oom even when they use Shadowfiend because it doesn't hit the target.

How it should work: Shadowfiend is a special pet, it teleports to the target whenever it gets kited for more than 1 second, therefore a priest can always get some mana back thanks to the pet. it's possible to stun him but it's impossible to kite the pet.
i played on Many pvp servers since wotlk came out. one of them is the most famous of all, Arena Tournament. and i can assure you that Shadowfiend is bugged in Sunwell.
 

Source (you should point out proofs of your report, please give us some source):  i give you this link to let you see how it works on the most scripted pvp server at this time, (Blackrock) one of warmane's realms.
Link : https://www.twitch.tv/videos/754551034
Also : Skip to 3:45 to see the begining of the test.

Thank you for your time.

**Comment by wonszopaionk** [comment]
Paymane is not relaiable source of anything. Provide a retail link with proof.

**Comment by mirailol** [comment]
First of all, thank you for your fast response.
but i since there are no wotlk servers in retail, i can try to ask a friend that plays live right now in the latest expansion to demonstrate how shadowfiend works. is that alright?

**Comment by Platinumcoated** [comment]
No, find a video from when retail was on WotLK showing this behaviour, or a wiki page from that time (using version history) describing it. Could also be an old wowhead comment or some old forum post. 
Some way that proves that this mechanic worked the way you describe back then, when WotLK was live.

**Comment by wonszopaionk** [comment]
What I've found:
![obraz](https://user-images.githubusercontent.com/28888570/94648112-1a35b280-02f2-11eb-9311-358a90e0227f.png)
![obraz](https://user-images.githubusercontent.com/28888570/94648286-8adccf00-02f2-11eb-8051-383ac7ef9cb4.png)
And an info that shadowcrawl has 6 sec CD which is just as in your video (1st shadowcrawl @ 3:58, 2nd shadowcrawl @ 4:04 = dt=6 sec; 3rd crawl @ 8:11, 4th crawl @8:17 dt=6 sec).
So it seems fine for me.

**Comment by mirailol** [comment]
Yes but in sunwell my pet gets kited forever and therefore doesn't land a single melee attack, which means after using shadowfiend, i didnt get a single drop of mana. and that happened many times in Arena.

**Comment by wonszopaionk** [comment]
#2965   #1425 
@mirailol check those bug reports, maybe there is still problem with pet movement speed (just like warlocks pets couldn't handle movement speed of bosses and players during movement)
#3291 #3075 #2825 
@Azathothh

**Comment by mirailol** [comment]
Alright. but just to remind you, priest pet is the only one that can teleport to its target, so by putting pet speed aside, i think the main probleme here is that it doesn't teleport to my target when it's being kited with frost trap, for example.

**Comment by wonszopaionk** [comment]
I suggest that you use a macro for your shadowfiend, like that:

#showtooltip Shadowfiend
/cast [nopet] Shadowfiend
/petaggressive
/petattack
/cast shadowcrawl
/cast [pet, target=pet] Dispel Magic
